miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultMetricService.java

```diff
@@ -67,21 +82,35 @@ protected void authorizeUpdate(Metric object) throws IOException {
     authorizationManager.throwIfNonAdmin();
   }
 
+  @Override
+  protected void loadChildEntities(Metric object) throws IOException {
+    loadChildEntity(object.getSubcategory(), object::setSubcategory, metricSubcategoryService);
+    loadChildEntity(object.getTissueMaterial(), object::setTissueMaterial, tissueMaterialService);
+    loadChildEntity(object.getTissueType(), object::setTissueType, tissueTypeService);
+    loadChildEntity(object.getTissueOrigin(), object::setTissueOrigin, tissueOriginService);
+    loadChildEntity(object.getContainerModel(), object::setContainerModel, containerModelService);
+  }
+
   @Override
   protected void collectValidationErrors(Metric object, Metric beforeChange, List<ValidationError> errors) throws IOException {
-    if ((ValidationUtils.isChanged(Metric::getAlias, object, beforeChange)
-        || ValidationUtils.isChanged(Metric::getCategory, object, beforeChange))
-        && metricDao.getByAliasAndCategory(object.getAlias(), object.getCategory()) != null) {
-      errors.add(ValidationError.forDuplicate("metric", "alias"));
-    }
+    // no checks
   }
 
   @Override
   protected void applyChanges(Metric to, Metric from) throws IOException {
     to.setAlias(from.getAlias());
-    to.setCategory(from.getCategory());
```

Comment:
Why remove the category assignment?

Reply:
Prevents changing category when subcategory is already set. Incompatible changes could be prevented with validation instead, but I think it would also be confusing if the category changed for metrics that are already linked to assays. This way, you have to delete and re-create the metric, which forces you to look at the assays it's linked to as well

Reply:
That sounds reasonable to me